scripts/xdcc_autoget.pl

```diff
@@ -0,0 +1,711 @@
+# made because torrents are watched by the feds, but xdcc lacks RSS feeds.
+# if you encounter any problems, fix it yourself you lazy bastard (or get me to), then contact me so I can add your fix and bump that version #
+# BeerWare License. Use any part you want, but buy me a beer if you ever meet me and liked this hacked together broken PoS
+# Somewhat based off of DCC multiget by Kaveh Moini.
+# USE: for help         : ag_help
+#      to run           : ag_run
+#      to halt at next  : ag_stop
+#      to set the server: ag_server 
+#      to add a bot     : ag_botadd BOT1 BOT2 *** BOTN
+#      to remove a bot  : ag_botrem BOT1 BOT2 *** BOTN
+#      to add string    : ag_add "[TEXT STRING OF TV SHOW/CHINESE CARTOON/ETC]","[ETC]",***,"[ETC]" 
+#      to remove strings: ag_rem "[TEXT STRING OF TV SHOW/CHINESE CARTOON/ETC]","[ETC]",***,"[ETC]" 
+#      ag_next_delay                 : delay between full transfers
+#      ag_dcc_closed_retry_delay     : delay after premature transfer
+#      ag_bot_delay: delay between request and when you "SHOULD" get it
+#      ag_interrun_delay             : delay (in minutes, the rest seconds) between finishing a round and starting another
+#      ag_autorun                    : whether to run on startup
+#      ag_xdcc_send_prefix           : the xdcc message before the pack #
+#      ag_xdcc_find_prefix           : the xdcc message before the search term
+#      ag_bot_file                   : where your bot list is stored
+#      ag_search_file                : where your search list is stored
+
+use strict;
+use warnings;
+
+use Irssi;
+use Text::ParseWords;
+use autodie; # die if problem reading or writing a file
+use File::HomeDir;
+use File::Copy;
+use Irssi 20090331;
+use vars qw($VERSION %IRSSI);
+
+$VERSION = 1.2;
+%IRSSI = (
+	name => "autoget", 
+	description => "XDCC Autoget, for automated searching and downloading of xdcc packs",
+	license => "BeerWare Version 42",
+	changed => "$VERSION",
+	authors => "MarshalMeatball",
+	contact => "mobilegundamseed\@hotmail.com",
+);
+
+Irssi::settings_add_int($IRSSI{'name'}, "ag_next_delay", 10);
+Irssi::settings_add_int($IRSSI{'name'}, "ag_dcc_closed_retry_delay", 10);
+Irssi::settings_add_int($IRSSI{'name'}, "ag_bot_delay", 30);
+Irssi::settings_add_int($IRSSI{'name'}, "ag_interrun_delay", 15);
+Irssi::settings_add_bool($IRSSI{'name'}, "ag_autorun", 1);
+Irssi::settings_add_bool($IRSSI{'name'}, "ag_episodic", 0);
+Irssi::settings_add_str($IRSSI{'name'}, "ag_xdcc_send_prefix", "xdcc send");
+Irssi::settings_add_str($IRSSI{'name'}, "ag_xdcc_cancel_prefix", "xdcc cancel");
+Irssi::settings_add_str($IRSSI{'name'}, "ag_find_prefix", "!find");
+Irssi::settings_add_str($IRSSI{'name'}, "ag_format", "");
+Irssi::settings_add_str($IRSSI{'name'}, "ag_folder", File::HomeDir->my_home);
+
+my @totags = ();	#timeout tags (need to be purged between send requests maybe)
+
+my $nexdelay = Irssi::settings_get_int("ag_next_delay");		#delay for next pack
```

Comment:
You need to make a `sub read_settings` (name does not matter) where you do all the
```
$variable = Irssi::settings_get_...
```
code. Then, you need to use
```
Irssi::signal_add('setup changed' => 'read_settings'); # do it whenever the user changes a setting with /set
read_settings(); # do it once on script load
```
Otherwise, the users of your script cannot change settings using `/set` (they would have to reload your script first, before changes take effect)